src/jit/emit.cpp

```diff
@@ -4684,20 +4706,16 @@ unsigned emitter::emitEndCodeGen(Compiler* comp,
 
 #define DEFAULT_CODE_BUFFER_INIT 0xcc
 
+    unsigned lastGroupPadding = 0;
+
     for (ig = emitIGlist; ig; ig = ig->igNext)
     {
         assert(!(ig->igFlags & IGF_PLACEHOLDER)); // There better not be any placeholder groups left
 
         /* Is this the first cold block? */
         if (ig == emitFirstColdIG)
         {
-            unsigned actualHotCodeSize = emitCurCodeOffs(cp);
```

Comment:
It is where we have added padding to the hot code block before, so because distance to the first coldIG is `emitTotalHotCodeSize`, not the `emitActualHotCodeSize` after we shrink hot block we have `lastHotIG->offset + lastHotIG->size < emitTotalHotCodeSize` and it is ` != firstColdIG->offset`. In order to maintain `lastHotIG->offset + lastHotIG->size ==  firstColdIG->offset` we need to increase `lastHotIG->size`.